File: apps/mobile/src/api/repositories/authRepository.ts
import { IAuthRepository } from './interfaces/IRepositories';
import { StudentProfile } from '../../store/useAuthStore';

export interface LoginParams {
  identifier: string;
  password: string;
}

export interface SignupParams {
  name: string;
  phoneOrEmail: string;
  password: string;
  classId: string;
}

class MockAuthRepository implements IAuthRepository {
  async login(_params: LoginParams): Promise<{ token: string; refreshToken?: string; user: StudentProfile }> {
    return {
      token: 'mock-jwt-token-123',
      refreshToken: 'mock-refresh-token-123',
      user: {
        id: 'student-1',
        name: 'রাফি আহমেদ',
        classId: 'class-8',
        className: 'Class 8',
        language: 'bn',
      },
    };
  }

  async signup(_params: SignupParams): Promise<{ status: string; referenceId?: string }> {
    return {
      status: 'OTP_SENT',
      referenceId: 'mock-ref-123',
    };
  }

  async verifyOtp(_referenceId: string, _otp: string): Promise<{ token: string; refreshToken?: string; user: StudentProfile }> {
    return {
      token: 'mock-jwt-token-123',
      refreshToken: 'mock-refresh-token-123',
      user: {
        id: 'student-1',
        name: 'রাফি আহমেদ',
        classId: 'class-8',
        className: 'Class 8',
        language: 'bn',
      },
    };
  }

  async getCurrentUser(): Promise<StudentProfile> {
    return {
      id: 'student-1',
      name: 'রাফি আহমেদ',
      classId: 'class-8',
      className: 'Class 8',
      language: 'bn',
    };
  }

  async refreshToken(_refreshToken: string): Promise<{ token: string; refreshToken?: string }> {
    return {
      token: 'mock-refreshed-jwt-token-456',
      refreshToken: 'mock-refreshed-refresh-token-456',
    };
  }

  async logout(): Promise<void> {
    return;
  }

  async updateProfile(updates: Partial<StudentProfile>): Promise<StudentProfile> {
    const current = await this.getCurrentUser();
    return { ...current, ...updates };
  }
}

export const authRepository = new MockAuthRepository();

File: apps/mobile/lib/repositories/auth_repository.dart
import '../core/network/http_client.dart';
import '../core/config/env.dart';
import '../models/user.dart';

class AuthRepository {
  Future<Map<String, dynamic>> login(String identifier, String password) async {
    if (ENV.useMockApi) {
      return {
        'token': 'mock-jwt-access-token',
        'refreshToken': 'mock-refresh-token',
        'user': StudentProfile(
          id: 'student-1',
          userId: 'user-1',
          name: 'শিক্ষার্থী',
          classId: '',
          className: '',
          language: 'bn',
        ).toJson(),
      };
    }
    return await httpClient.request('/auth/login',
        method: 'POST',
        body: {
          'identifier': identifier,
          'password': password,
        },
        skipAuth: true);
  }

  Future<Map<String, dynamic>> signup(
      String name, String phoneOrEmail, String password) async {
    if (ENV.useMockApi) {
      return {'status': 'OTP_SENT', 'referenceId': 'mock-ref-123'};
    }
    return await httpClient.request('/auth/signup',
        method: 'POST',
        body: {
          'name': name,
          'phoneOrEmail': phoneOrEmail,
          'password': password,
        },
        skipAuth: true);
  }

  Future<Map<String, dynamic>> verifyOtp(String referenceId, String otp) async {
    if (ENV.useMockApi) {
      return {
        'token': 'mock-jwt-access-token',
        'refreshToken': 'mock-refresh-token',
        'user': StudentProfile(
          id: 'student-1',
          userId: 'user-1',
          name: 'শিক্ষার্থী',
          classId: '',
          className: '',
          language: 'bn',
        ).toJson(),
      };
    }
    return await httpClient.request('/auth/verify-otp',
        method: 'POST',
        body: {
          'referenceId': referenceId,
          'otp': otp,
        },
        skipAuth: true);
  }

  Future<StudentProfile> getCurrentUser() async {
    if (ENV.useMockApi) {
      return StudentProfile(
        id: 'student-1',
        userId: 'user-1',
        name: 'শিক্ষার্থী',
        classId: '',
        className: '',
        language: 'bn',
      );
    }
    final json = await httpClient.request('/auth/me');
    return StudentProfile.fromJson(json);
  }

  Future<void> logout() async {
    if (!ENV.useMockApi) {
      try {
        await httpClient.request('/auth/logout', method: 'POST');
      } catch (_) {}
    }
  }
}

final authRepository = AuthRepository();
